### PR TITLE
Added two multimodal experiments: MathVista and ScienceQA

### PR DESCRIPTION
**Pull request by lupantech**
Updates:
- Added `textgrad/tasks/multimodal/`
- Added `evaluation/results/solution_optimization_mm/`
- Added `examples/mathvista/` and `examples/notebooks/TextGrad-Vision-MathVista.ipynb`

### CODE REVIEW

File: examples/notebooks/TextGrad-Vision.ipynb
Comment (vinid) [inline examples/notebooks/TextGrad-Vision.ipynb:1]: I think this was renamed in the latest PR. We can probably remove this unless you made any exit to this one

**Review by vinid — APPROVED** [review]

File: examples/notebooks/TextGrad-Vision.ipynb
Comment (lupantech) [reply examples/notebooks/TextGrad-Vision.ipynb:1]: Deleted `TextGrad-Vision.ipynb`.

File: evaluation/solution_optimization_mm.py [hunk 1]
```diff
@@ -0,0 +1,285 @@
...
+            if args.test_num > 0 and i >= args.test_num:
+                break
+            future = executor.submit(run_test_time_training, sample)
+            futures.append(future)
+
+        print("\nRunning test-time training...")
+        for _, future in enumerate(tqdm(concurrent.futures.as_completed(futures), total=len(futures), position=0)):
+            question, answer, predictions, performance_history, loss_history, result_data, ques_data = future.result() # NOTE: check the return format
+            pid = ques_data["pid"] # NOTE: check the question id
+            all_solutions[pid] = {
```
Comment (mertyg) [inline evaluation/solution_optimization_mm.py:215]: can we just remove the case where args.num_threads is not 0? This is just to make the script a bit more concise. no worries if not.

File: evaluation/results/solution_optimization_mm/scienceqa_predictions.json
Comment (mertyg) [inline evaluation/results/solution_optimization_mm/scienceqa_predictions.json:1]: propose to move these to a gdrive folder

File: evaluation/results/solution_optimization_mm/mathvista_predictions.json
Comment (mertyg) [inline evaluation/results/solution_optimization_mm/mathvista_predictions.json:1]: propose to move these to a gdrive folder

**Review by mertyg — APPROVED** [review]

**Pull request merged**